src/requests/actions.jl

```diff
@@ -7,7 +7,7 @@ end
 function textDocument_codeAction_request(params::CodeActionParams, server::LanguageServerInstance, conn)
     commands = Command[]
     doc = getdocument(server, URI2(params.textDocument.uri))
-    offset = get_offset(doc, params.range.start) # Should usef get_offset2?
+    offset = get_offset2(doc, params.range.start)
     x = get_expr(getcst(doc), offset)
     arguments = Any[params.textDocument.uri, offset] # use the same arguments for all commands
```

Comment:
So, I never understood what exactly `get_offset` returns, but I always thought that it was _the thing_ that things like `get_expr` expect...

`get_offset2` returns a valid index into the underlying string that holds the document, and I primarily introduced it to fix the text sync implementation. But I'm not sure we can just swap out `getoffset2` for `getoffset` in situations where we then pass the return value onto functionality that was built assuming the `get_offset` semantics?

But take this just as a heads-up about the history, this fix might also just be correct, I don't really know anything about `get_expr` or how it works :)

Reply:
Honestly I didn't look into the details, but while debugging I noticed the comment and just tried it out. It seems to fix all the issues I had at least.